src/metabase/driver/postgres.clj

```diff
@@ -296,6 +300,44 @@
   [driver [_ arg]]
   (sql.qp/->honeysql driver [:percentile arg 0.5]))
 
+(defmethod sql.qp/->honeysql [:postgres :datediff]
```

Comment:
postgres works the following:
- year: `date_part( year, age(x, y))`
- week: `datediff(x, y, day) / 7`
- month: `datediff(x, y, year) * 12 + date_part( month, age(x, y))`
- day: `date_part( day, x - y)`
- hour: `datediff(x, y, day) * 24 + date_part( hour, age(x, y))`
- minute, and second are similar to hour and month where they get the datediff for the coarser resolution, multiply by 60, and then get the date_part from the age.